### PR TITLE
[dist] update base image for deploy container

### PR DESCRIPTION
**Pull request by PratikMahajan**

### CODE REVIEW

**Comment by petr-muller** [comment]
```
 [1/3] STEP 8/9: RUN mkdir -p $HOME/.cargo/git/ &&   find $HOME/. -type d -exec chmod 777 {} \; &&   find $HOME/. -type f -exec chmod ugo+rw {} \;
[1/3] STEP 9/9: RUN hack/build_e2e.sh
/bin/sh: line 1: hack/build_e2e.sh: No such file or directory 
```

We do not expect that, right?

**Review by petr-muller — APPROVED** [review]
Changes LGTM, tests should pass after https://github.com/openshift/release/pull/36841

/hold

**Comment by petr-muller** [comment]
```
/bin/sh: dnf: command not found
```
at least we got further ;)

**Comment by PratikMahajan** [comment]
/override ci/prow/customrust-e2e
/override ci/prow/customrust-images

it'll be fixed in #786

File: dist/Dockerfile.deploy/Dockerfile
Comment (petr-muller) [inline dist/Dockerfile.deploy/Dockerfile]: What is `redhat/ubi8:latest` supposed to resolve to? This should be either just `ubi8` or its long form `registry.access.redhat.com/ubi8`. If I just do `podman build . -f dist/Dockerfile.deploy/Dockerfile` with the file with `redhat/ubi8:latest` then podman asks me what to resolve to and it treats `redhat` as a repository in =e.g. quay:

```
$ podman build . -f dist/Dockerfile.deploy/Dockerfile
[1/2] STEP 1/10: FROM redhat/ubi8:latest AS builder
? Please select an image: 
    registry.fedoraproject.org/redhat/ubi8:latest
    registry.access.redhat.com/redhat/ubi8:latest
  ▸ docker.io/redhat/ubi8:latest
    quay.io/redhat/ubi8:latest
```

With `FROM registry.access.redhat.com/ubi8:latest` the build goes just fine.

Comment (PratikMahajan) [reply dist/Dockerfile.deploy/Dockerfile]: fixed it to use registry.access

**Review by petr-muller — APPROVED** [review]
This builds for me but will need overrides, I guess

**Pull request merged**
